### PR TITLE
[RISCV] Support Strict FP arithmetic Op when only have Zvfhmin

### PR DESCRIPTION
**Pull request by jacquesguan**
Include: STRICT_FADD, STRICT_FSUB, STRICT_FMUL, STRICT_FDIV, STRICT_FSQRT and STRICT_FMA.

### CODE REVIEW

**Comment by jacquesguan** [comment]
ping...

File: llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp [hunk 1]
```diff
@@ -634,6 +636,38 @@ void VectorLegalizer::PromoteSETCC(SDNode *Node,
...
+  SDLoc DL(Node);
+  SmallVector<SDValue, 5> Operands(Node->getNumOperands());
+
+  for (unsigned j = 0; j != Node->getNumOperands(); ++j)
+    if (Node->getOperand(j).getValueType().isVector() &&
+        !(ISD::isVPOpcode(Node->getOpcode()) &&
+          ISD::getVPMaskIdx(Node->getOpcode()) == j)) // Skip mask operand.
+      // promote the vector operand.
+      Operands[j] =
+          DAG.getNode(ISD::FP_EXTEND, DL, NewVecVT, Node->getOperand(j));
```
Comment (topperc) [inline llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp:655]: Why aren't the inserted nodes also STRICT?

Comment (jacquesguan) [reply llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp:655]: Should we use the STRICT extend or round in promotion? I think only the calculation op is constrained. This is not clear for me.

Comment (topperc) [reply llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp:655]: The Promote code in LegalizeDAG.cpp uses STRICT_FP_EXTEND and STRICT_FP_ROUND.

Comment (jacquesguan) [reply llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp]: Done, changed to STRICT_FP_ROUND and STRICT_FP_EXTEND.

**Review by michaelmaitland — APPROVED** [review]
LGTM

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**